server/src/test/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequestTests.java

```diff
@@ -131,4 +137,52 @@ public void testValidateErrorMessage() throws Exception {
         assertThat(noError, is(nullValue()));
     }
 
+    private PutIndexTemplateRequest randomPutIndexTemplateRequest() throws IOException {
+        PutIndexTemplateRequest request = new PutIndexTemplateRequest();
+        request.name("test");
+        if (randomBoolean()){
+            request.version(randomInt());
+        }
+        if (randomBoolean()){
+            request.order(randomInt());
+        }
+        request.patterns(Arrays.asList(generateRandomStringArray(20, 100, false, false)));
+        int numAlias = between(0, 5);
+        for (int i = 0; i < numAlias; i++) {
+            Alias alias = new Alias(randomRealisticUnicodeOfLengthBetween(1, 10));
+            if (randomBoolean()) {
+                alias.indexRouting(randomRealisticUnicodeOfLengthBetween(1, 10));
+            }
+            if (randomBoolean()) {
+                alias.searchRouting(randomRealisticUnicodeOfLengthBetween(1, 10));
+            }
+            request.alias(alias);
+        }
+        if (randomBoolean()) {
+            request.mapping("doc", XContentFactory.jsonBuilder().startObject()
+                .startObject("doc").startObject("properties")
+                .startObject("field-" + randomInt()).field("type", randomFrom("keyword", "text")).endObject()
+                .endObject().endObject().endObject());
+        }
+        if (randomBoolean()){
+            request.settings(Settings.builder().put("setting1", randomLong()).put("setting2", randomTimeValue()).build());
+        }
+        return request;
+    }
+
+    public void testFromToXContentPutTemplateRequest() throws Exception {
+        for (int i = 0; i < 10; i++) {
+            PutIndexTemplateRequest expected = randomPutIndexTemplateRequest();
+            XContentType xContentType = randomFrom(XContentType.values());
+            BytesReference shuffled = toShuffledXContent(expected, xContentType, ToXContent.EMPTY_PARAMS, randomBoolean());
+            PutIndexTemplateRequest parsed = new PutIndexTemplateRequest().source(shuffled, xContentType);
+            assertNotSame(expected, parsed);
+            assertThat(parsed.version(), equalTo(expected.version()));
+            assertThat(parsed.order(), equalTo(expected.order()));
+            assertThat(parsed.patterns(), equalTo(expected.patterns()));
+            assertThat(parsed.aliases(), equalTo(expected.aliases()));
+            assertThat(parsed.mappings(), equalTo(expected.mappings()));
+            assertThat(parsed.settings(), equalTo(expected.settings()));
+        }
```

Comment:
I would make this class extend `AbstractXContentTestCase`, then your randomPutIndexTemplateRequest would become `createTestInstance`, and

```
    @Override
    protected PutIndexTemplateRequest doParseInstance(XContentParser parser) throws IOException {
        return new PutIndexTemplateRequest().source(parser.map());
    }

    @Override
    protected boolean supportsUnknownFields() {
        return false;
    }

    @Override
    protected void assertEqualInstances(PutIndexTemplateRequest expected, PutIndexTemplateRequest parsed) {
        assertNotSame(expected, parsed);
        assertThat(parsed.version(), equalTo(expected.version()));
        assertThat(parsed.order(), equalTo(expected.order()));
        assertThat(parsed.patterns(), equalTo(expected.patterns()));
        assertThat(parsed.aliases(), equalTo(expected.aliases()));
        assertThat(parsed.mappings(), equalTo(expected.mappings()));
        assertThat(parsed.settings(), equalTo(expected.settings()));
    }
```